### PR TITLE
style(beta): use logical properties

### PR DESCRIPTION
**Pull request by 321gillian**
Changing beta components to logical properties

✅ Closes: COMUI-3096

### CODE REVIEW

**Comment by daragh-king-genesys** [comment]
I would have expected to see a change [here](https://github.com/MyPureCloud/genesys-spark/blob/main/packages/genesys-spark-components/.stylelintrc.json) and then running the linter sort out the actual scss files

**Comment by 321gillian** [comment]
Some issues here with popovers and the toggle. Looking into them.

File: packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.scss [hunk 1]
```diff
@@ -137,6 +138,12 @@ gux-toggle-slider {
           }
         }
       }
+
+      &.gux-rtl {
+        .gux-switch {
+          transform: translateX($slider-transition-rtl);
+        }
+      }
```
Comment (321gillian) [inline packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.scss:146]: manual change

File: packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.tsx [hunk 2]
```diff
@@ -32,6 +35,10 @@ export class GuxToggleSlider {
     if (this.errorId) {
       this.checkboxElement?.setAttribute('aria-describedby', this.errorId);
     }
+
+    // TODO: Use :dir() selector instead when Chrome improves support for it
+    const compStyles = window.getComputedStyle(this.checkboxElement);
+    this.direction = compStyles.direction;
```
Comment (321gillian) [inline packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.tsx:41]: flagging manual changes in this file

File: packages/genesys-spark-components/.stylelintrc.json [hunk 3]
```diff
@@ -1,15 +1,15 @@
 {
   "extends": ["stylelint-config-genesys-spark-components"],
+  "plugins": ["stylelint-use-logical-spec"],
+  "rules": {
+    "liberty/use-logical-spec": true
+  },
   "overrides": [
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/.stylelintrc.json:7]: This overrides section can be removed

File: packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.scss (hunk 1 above)
Comment (321gillian) [reply packages/genesys-spark-components/src/components/stable/gux-toggle/gux-toggle-slider/gux-toggle-slider.scss:146]: This is what it looks like without it. Prior to the changes to logical properties it looked fine just not mirrored. 

https://github.com/user-attachments/assets/5bb179be-42e9-4b10-950c-0a8d69cfeac7

File: packages/genesys-spark-components/.stylelintrc.json [hunk 4]
```diff
@@ -1,15 +1,7 @@
...
-      "plugins": ["stylelint-use-logical-spec"],
-      "rules": {
-        "liberty/use-logical-spec": true
-      }
-    }
-  ]
+  "plugins": ["stylelint-use-logical-spec"],
+  "rules": {
+    "liberty/use-logical-spec": true
+  }
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/.stylelintrc.json:6]: Looking at this PR again I think we should do this in a few PRs over a few weeks to reduce the risk. We are also not going to want to change the legacy components at all so we will need to have an override for that.

See can you find a config that changes about 40 files. Also have the example.html files changes in their own PR as they are not that critical.

**Review by daragh-king-genesys — CHANGES_REQUESTED** [review]

File: packages/genesys-spark-components/.stylelintrc.json (hunk 4 above)
Comment (321gillian) [reply packages/genesys-spark-components/.stylelintrc.json]: Was thinking it might be good to start with beta components and go from there.

**Review by katie-bobbe-genesys — APPROVED** [review]
I took a look at the beta components in different browsers and I did not see any regressions.

**Comment by jason-evans-genesys** [comment]
I tested all the components that this PR touched and I also did not see any regressions.

**Review by jason-evans-genesys — APPROVED** [review]

**Comment by daragh-king-genesys** [comment]
@321gillian you can get this PR ready again and we can get it merged

**Review by daragh-king-genesys — APPROVED** [review]

**Pull request merged**